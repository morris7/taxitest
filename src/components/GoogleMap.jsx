import React, { Component } from 'react';
import { Map, GoogleApiWrapper } from 'google-maps-react';
import PropTypes from 'prop-types';

class GoogleMapContainer extends Component {

  constructor(props){
    super(props);
  }

  render(){
    const mapStyles = {
      width: '1000px',
      height: '600px'
    };

    return (
      <Map
         google={this.props.google}
         zoom={14}
         style={mapStyles}
         initialCenter={{lat: 51.505, lng: -0.090}}
       >
       {this.props.markers}
      </Map>
     );
  }
}

GoogleMapContainer.propTypes = {
  markers: PropTypes.array
}

export default GoogleApiWrapper({
  apiKey: 'AIzaSyDqPjL56MrI0i6noAg82tbix3EjJGEE2lM'
})(GoogleMapContainer);
