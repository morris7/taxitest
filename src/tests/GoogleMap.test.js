import React from 'react';
import ReactDOM from 'react-dom';
import GoogleMap from '../components/GoogleMap';
const ReactTestRenderer = require('react-test-renderer');

describe('when rendering a GoogleMap component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<GoogleMap />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const component = ReactTestRenderer.create(<GoogleMap />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
