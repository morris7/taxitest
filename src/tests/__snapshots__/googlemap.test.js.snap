// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering a GoogleMap component should match snapshot 1`] = `
<div>
  Loading...
</div>
`;
